src/Illuminate/Log/Context/ContextServiceProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Illuminate\Log\Context;
+
+use Illuminate\Log\Context\Events\Dehydrating;
+use Illuminate\Log\Context\Events\Hydrated;
+use Illuminate\Queue\Events\JobProcessing;
+use Illuminate\Queue\Queue;
+use Illuminate\Support\ServiceProvider;
+
+class ContextServiceProvider extends ServiceProvider
+{
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->app->singleton(Repository::class);
+    }
+
+    /**
+     * Boot the application services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        Queue::createPayloadUsing(function ($c, $q, $payload) {
+            $context = $this->app[Repository::class];
+
+            $this->app['events']->dispatch(new Dehydrating($context));
+
+            if ($context->all() === [] && $context->allSecrets() === []) {
+                return $payload;
+            }
+
+            return [
+                ...$payload,
+                '__laravel_context' => [
+                    'data' => $context->all(),
+                    'secrets' => $context->allSecrets(),
+                ],
+            ];
+        });
+
+        $this->app['events']->listen(function (JobProcessing $event) {
+            $context = $this->app[Repository::class]->flush();
+
+            [
+                'data' => $data,
+                'secrets' => $secrets,
+            ] = $event->job->payload()['__laravel_context'] ?? ['data' => [], 'secrets' => []];
+
+            $context->add($data)->addSecret($secrets);
+
+            $this->app['events']->dispatch(new Hydrated($context));
+        });
+
+        // TODO move to Octane
+        $this->app['events']->listen([
+            \Laravel\Octane\Events\RequestReceived::class,
+            \Laravel\Octane\Events\TaskReceived::class,
+            \Laravel\Octane\Events\TickReceived::class,
+        ], fn ($event) => $event->sandbox[Repository::class]->flush()->setApplication($event->sandbox));
```

Comment:
I will move this to Octane before we merge.

It is here to make testing easier for any reviewers.